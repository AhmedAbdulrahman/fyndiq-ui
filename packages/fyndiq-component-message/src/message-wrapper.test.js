import React from 'react'
import { shallow } from 'enzyme'

import Message from './message'
import Wrapper from './message-wrapper'

describe('fyndiq-component-message MessageWrapper', () => {
  beforeEach(() => {
    Wrapper.instance = null
  })

  it('should keep an instance of the running wrapper', () => {
    shallow(<Wrapper />)
    expect(Wrapper.instance).not.toBe(null)
  })

  it('should warn the user if there are 2 instances of wrapper', () => {
    console.warn = jest.fn()
    shallow(<Wrapper />)
    shallow(<Wrapper />)
    expect(console.warn).toHaveBeenCalled()
  })

  it('should remove messages internally', () => {
    const component = shallow(<Wrapper />)
    Wrapper.addMessage(<Message>New message</Message>)
    component.update()
    component.find(Message).simulate('close')
    component.update()
    expect(component.find(Message).exists()).toBe(false)
  })

  it('should not render anything if there is no message', () => {
    expect(shallow(<Wrapper />).isEmptyRender()).toBe(true)
  })

  describe('static addMessage', () => {
    it('should have a static addMessage function', () => {
      const component = shallow(<Wrapper />)
      Wrapper.addMessage(<Message>New message</Message>)
      component.update()
      expect(component.find(Message).exists()).toBe(true)
    })

    it('should throw if there is no instance of Wrapper', () => {
      expect(() => Wrapper.addMessage(<Message>New message</Message>)).toThrow()
    })

    it('should have the proper structure', () => {
      Math.random = jest.fn(() => 'RANDOM-ID')
      const component = shallow(<Wrapper />)
      Wrapper.addMessage(<Message>New message 1</Message>)
      Wrapper.addMessage(<Message>New message 2</Message>)
      component.update()
      expect(component).toMatchSnapshot()
    })
  })
})
