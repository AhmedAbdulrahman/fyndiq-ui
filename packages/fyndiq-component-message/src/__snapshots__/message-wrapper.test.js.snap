// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-message MessageWrapper static addMessage should have the proper structure 1`] = `
<MessagePortal
  portalId="fyndiq-message-portal"
>
  <div
    className="wrapper"
  >
    <div
      key="RANDOM-ID"
    >
      <Message
        icon={null}
        id="RANDOM-ID"
        onClose={[Function]}
        timeout={5000}
        type="info"
      >
        New message 1
      </Message>
    </div>
    <div
      key="RANDOM-ID"
    >
      <Message
        icon={null}
        id="RANDOM-ID"
        onClose={[Function]}
        timeout={5000}
        type="info"
      >
        New message 2
      </Message>
    </div>
  </div>
</MessagePortal>
`;
